### PR TITLE
Fixes #17805 - update to rubocop 0.46

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -26,6 +26,14 @@ SpaceInsideHashLiteralBraces:
 HashSyntax:
   Enabled: false # don't force 1.9 hash syntax
 
+Rails/DynamicFindBy:
+  Enabled: False
+  Whitelist:
+    - find_by_cp_id
+    - find_by_katello_id
+    - find_by_unit_id
+    - find_by_version
```
Comment (jlsherrill) [inline .rubocop.yml]: opened http://projects.theforeman.org/issues/17806  to resolve this

**Comment by jlsherrill** [comment]
CI run: http://ci.theforeman.org/job/test_katello_pull_request/10799/

**Comment by akofink** [comment]
LGTM! APJ

**Comment by jlsherrill** [comment]
Build is green, merging!

**Pull request merged**